lib/models/sport_session.rb

```diff
@@ -1,19 +1,35 @@
 # frozen_string_literal: true
 
 module Models
-  class SportSession
-    attr_accessor :id, :notes,
-                  :distance, :trace, :duration, :start_time, :end_time, :timezone, :start_time_timezone_offset,
-                  :year, :month,
-                  :elevation_gain, :elevation_loss, :sport_type_id,
-                  :pause, :sport_type
-
-    def initialize(attrs = {})
-      attrs.each do |k, v|
-        send("#{k}=", v)
-      end
+  class SportSession < Definition::ValueObject
+    class GpsPointArray < Definition::ValueObject
+      definition(Definition.Each(Definition.Keys do
+        required 'time', Definition.Type(Time)
+        required 'lat', Definition.Type(Float)
+        required 'lng', Definition.Type(Float)
+        optional 'ele', Definition.Type(Float)
+      end))
     end
 
+    definition(Definition.Keys do
+      required :id, Definition.Type(String)
+      optional :sport_type_id, Definition.Type(Integer)
+      optional :sport_type, Definition.Type(String)
```

Comment:
I prefer having the sport_type as a hash/value object with `id` and `name` fields.

Reply:
I am not sure if we should not just remove the sport_type_id .. is it necessary ?

Reply:
The sport_type_id is used to fetch the activities when searching by sport_type. But as far as I remembered it's not necessary to keep it in the model.

Reply:
Yes but all the sport-type-id usages can easily be changed to just use "sport_type", no need for an integer id imo.

Suggestion: keep it like this right here now, and i prepare another PR based on this (using definition already) without the sport-type-id so we see if this works out properly or not :-)